cli/src/update.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{
+    config::{Config, ReleaseInfo},
+    error::WKCliError,
+    output::colored_print,
+    utils::compare_with_current_time,
+};
+use aion::*;
+use chrono::Utc;
+use clap::crate_version;
+use log::debug;
+use owo_colors::OwoColorize;
+use reqwest::Client;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+
+const WUKONG_GITHUB_REPO: &str = "mindvalley/wukong-cli";
+const GITHUB_API_URL: &str = "https://api.github.com";
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GithubLatestReleaseInfo {
+    pub tag_name: String,
+    pub url: String,
+    pub published_at: String,
+    pub html_url: String,
+}
+
+// check_for_update checks whether this wukong has had a newer release on Github
+pub async fn check_for_update() -> Result<(), WKCliError> {
+    let release_info = get_current_release_info()?;
+
+    if let Some(release_info) = release_info {
+        let last_update_checked_since =
+            compare_with_current_time(&release_info.checked_for_update_at);
+
+        if last_update_checked_since >= -(24.hours()) {
```

Comment:
I believe the ticket and your PR description mentions `To avoid spam, we should only display this maximum 1 times per week.`, so this line is wrong